raiden/transfer/state.py

```diff
@@ -908,7 +908,10 @@ def __init__(
         if not isinstance(participant_address, typing.T_Address):
             raise ValueError('participant_address must be of type address')
 
-        if not isinstance(contract_balance, typing.T_BlockNumber):
+        if not isinstance(contract_balance, typing.T_TokenAmount):
+            raise ValueError('contract_balance must be of type token_amount')
+
+        if not isinstance(deposit_block_number, typing.T_BlockNumber):
```

Comment:
Change the following `ValueError` to properly report `deposit_block_number` as the offending argument.